dashboards_bundlers/server_upload/__init__.py

```diff
@@ -18,42 +22,59 @@ def bundle(handler, abs_nb_path):
     notebook_basename = os.path.basename(abs_nb_path)
     notebook_name = os.path.splitext(notebook_basename)[0]
 
-    # Make information about the request Host header available for use in
-    # constructing the urls
-    segs = handler.request.host.split(':')
-    hostname = segs[0]
-    if len(segs) > 1:
-        port = segs[1]
-    else:
-        port = ''
-    protocol = handler.request.protocol
-
-    # Treat empty as undefined
-    dashboard_server = os.getenv('DASHBOARD_SERVER_URL')
-    if dashboard_server:
-        dashboard_server = dashboard_server.format(protocol=protocol,
-            hostname=hostname, port=port)
-        upload_url = url_path_join(dashboard_server, UPLOAD_ENDPOINT,
-            escape.url_escape(notebook_name, False))
-        with open(abs_nb_path, 'rb') as notebook:
-            headers = {}
-            token = os.getenv('DASHBOARD_SERVER_AUTH_TOKEN')
-            if token:
-                # TODO: server side should expect Authorization: token <value>
-                headers['Authorization'] = token
-            result = requests.post(upload_url, files={'file': notebook},
-                headers=headers, timeout=60)
-            if result.status_code >= 400:
-                raise web.HTTPError(result.status_code)
-
-        # Redirect for client might be different from internal upload URL
-        redirect_server = os.getenv('DASHBOARD_REDIRECT_URL')
-        if redirect_server:
-            redirect_root = redirect_server.format(hostname=hostname,
-                port=port, protocol=protocol)
+    # Python 2/3 compatible try/finally to cleanup a temp working directory
+    tmp_dir = tempfile.mkdtemp()
+    try:
+        output_dir = os.path.join(tmp_dir, notebook_name)
+
+        # Clean up bundle dir if it exists
+        shutil.rmtree(output_dir, True)
+        os.makedirs(output_dir)
+
+        # Include frontend files referenced via the jupyter_cms bundle mechanism
+        shutil.copy2(abs_nb_path, os.path.join(output_dir, 'index.ipynb'))
+        bundle_file_references(output_dir, abs_nb_path, handler.tools)
+        bundle_declarative_widgets(output_dir, abs_nb_path)
+        bundle_web_static(output_dir)
```

Comment:
Right. At most, the zip should contain the notebook, notebook files (images, etc), and bower components for declwidgets.  There's the question about whether to include the declwidgets impl itself, since that is tied to some of the components in the bower directory. So I would say to include that as well (if possible).